src/plonkish/compiler/mod.rs

```diff
@@ -95,10 +95,16 @@ pub fn compile_phase1<
         )
     });
 
+    unit.compilation_phase = 1;
+
     (unit, assignment)
 }
 
 pub fn compile_phase2<F: Field + Clone>(unit: &mut CompilationUnit<F>) {
+    if unit.compilation_phase != 1 {
```

Comment:
@rutefig we could also panic if you try to do phase1 twice

Reply:
checking the implications of that